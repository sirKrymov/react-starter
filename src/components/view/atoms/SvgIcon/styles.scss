@import '../../../../styles/mixins';

.svg-icon {
  > div {
    @include center;
  }

  &--size-xs {
    @include svg-size(10px);
  }

  &--size-sm {
    @include svg-size(12px);
  }

  &--size-md {
    @include svg-size(16px);
  }

  &--size-lg {
    @include svg-size(20px);
  }

  &--size-xl {
    @include svg-size(24px);
  }

  &--size-xxl {
    @include svg-size(32px);
  }

  &--size-xxxl {
    @include svg-size(40px);
  }

  &--color-primary {
    @include svg-color(var(--primary));
  }

  &--color-on-primary {
    @include svg-color(var(--onPrimary));
  }

  &--color-secondary {
    @include svg-color(var(--secondary));
  }

  &--color-thirdy {
    @include svg-color(var(--thirdy));
  }

  &--color-valid {
    @include svg-color(var(--valid));
  }

  &--color-error {
    @include svg-color(var(--error));
  }

  &--rotate-90 {
    transform: rotate(90deg);
  }

  &--rotate-180 {
    transform: rotate(180deg);
  }

  &--rotate-270 {
    transform: rotate(270deg);
  }

  &--rotate-360 {
    transform: rotate(360deg);
  }

  &--top-1 {
    margin-top: 1px;
  }

  &--top-2 {
    margin-top: 2px;
  }

  &--top-3 {
    margin-top: 3px;
  }

  &--top-4 {
    margin-top: 4px;
  }

  &--pointer {
    cursor: pointer;
  }
}
