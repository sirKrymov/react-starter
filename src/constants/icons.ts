export { default as visibilityOffIcon } from 'assets/icons/svg/visibility_off.svg';
export { default as visibilityOnIcon } from 'assets/icons/svg/visibility_on.svg';
export { default as eyeCrossedIcon } from 'assets/icons/svg/eye_crossed.svg';
export { default as eyeIcon } from 'assets/icons/svg/eye.svg';
