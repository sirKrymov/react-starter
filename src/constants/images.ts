export { default as krymovLogo } from 'assets/images/krymov-logo.png';
