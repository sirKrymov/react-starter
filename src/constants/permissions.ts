import { IPagesPermissions } from 'types/permissions.interface';

export enum UserRoles {
  SUPERADMINISTRATOR = 'SUPERADMINISTRATOR',
  ADMINISTRATOR = 'ADMINISTRATOR',
  OPERATOR = 'OPERATOR'
}

export const pagePermissions: IPagesPermissions = {
  weather: [UserRoles.SUPERADMINISTRATOR, UserRoles.ADMINISTRATOR]
};
