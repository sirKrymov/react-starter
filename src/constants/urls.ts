export default {
  auth: {
    resetPassword: '/auth/reset-password',
    newPassword: '/auth/new-password',
    signin: '/auth/signin',
    signup: '/auth/signup'
  },

  main: {
    dashboard: '/dashboard',
    weather: '/weather',
    chat: '/chat'
  },

  auxiliary: {
    components: '/components'
  }
};
