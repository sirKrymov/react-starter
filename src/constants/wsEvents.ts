export enum WebSocketEvents {
  INVALID_TOKEN = 'InvalidToken',
  SYSTEM_STATUS = 'SystemStatus',
  DISCONNECT = 'disconnect',
  CONNECT = 'connect',
  MESSAGE = 'message',
  ERROR = 'Error'
}
