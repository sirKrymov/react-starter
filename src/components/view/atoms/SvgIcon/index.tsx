// @SETUP: Please don't forget to remove this component or current comment if component is used in project
import React, { ReactElement } from 'react';

import { ReactSVG } from 'react-svg';
import cn from 'classnames';

import './styles.scss';

interface IProps {
  className?: string;
  onClick?(): void;
  pointer?: boolean;
  rotate?: '90' | '180' | '270' | '360';
  color?: 'on-primary' | 'secondary' | 'primary' | 'thirdy' | 'valid' | 'error';
  size?: 'xs' | 'sm' | 'md' | 'lg' | 'xl' | 'xxl' | 'xxxl';
  src: string;
  top?: '1' | '2' | '3' | '4';
  id?: string;
}

export function SvgIcon({
  className,
  pointer,
  onClick,
  rotate,
  color,
  size,
  src,
  top,
  id
}: IProps): ReactElement {
  const svgIconClass = cn(
    {
      'svg-icon': true,
      [`svg-icon--rotate-${rotate}`]: rotate,
      [`svg-icon--color-${color}`]: color,
      [`svg-icon--size-${size}`]: size,
      [`svg-icon--top-${top}`]: top,
      'svg-icon--pointer': pointer
    },
    className
  );

  return (
    <ReactSVG src={src} className={svgIconClass} onClick={onClick} id={id} />
  );
}
