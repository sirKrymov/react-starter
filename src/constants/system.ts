import urls from './urls';

export const REFRESH_TOKEN_NAME = 'Refresh';
export const TOKEN_NAME = 'Bearer';
export const redirect = {
  AUTH: urls.main.dashboard,
  NOT_AUTH: urls.auth.signin
};

// For starter-kit testing
export const OPEN_WEATHER_API_KEY = '85b1af13ffddfd8e6fd69752f43a412a';
