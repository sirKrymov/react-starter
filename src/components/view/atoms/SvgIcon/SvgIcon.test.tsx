import React from 'react';

import { shallow, ShallowWrapper } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';

import { SvgIcon } from '../SvgIcon';

import { eyeCrossedIcon } from 'constants/icons';

describe('<SvgIcon> atom component', () => {
  let shallowComponent: ShallowWrapper;

  const createComponent = () => {
    shallowComponent = shallow(<SvgIcon src={eyeCrossedIcon} />);
  };

  describe('WHEN the component loads', () => {
    beforeEach(() => {
      createComponent();
    });

    it('THEN it should render without crashing', () => {
      expect(shallowComponent.html()).not.toHaveLength(0);
    });

    it('THEN it should match snapshot', () => {
      expect(shallowToJson(shallowComponent)).toMatchSnapshot();
    });
  });
});
