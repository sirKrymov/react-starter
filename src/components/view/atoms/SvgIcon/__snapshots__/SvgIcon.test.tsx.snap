// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SvgIcon> atom component WHEN the component loads THEN it should match snapshot 1`] = `
<ReactSVG
  afterInjection={[Function]}
  beforeInjection={[Function]}
  className="svg-icon"
  evalScripts="never"
  fallback={null}
  loading={null}
  renumerateIRIElements={true}
  src="eye_crossed.svg"
  wrapper="div"
/>
`;
